services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1865,6 +1865,61 @@ public void setStoreCurrentVersion(String clusterName, String storeName, int ver
             + "setting version on parent is not supported, since the version list could be different fabric by fabric");
   }
 
+  @Override
+  public void rollForwardToFutureVersion(String clusterName, String storeName) {
+    acquireAdminMessageLock(clusterName, storeName);
+    try {
+      getVeniceHelixAdmin().checkPreConditionForUpdateStoreMetadata(clusterName, storeName);
+      // Call child controllers in parallel to check whether backup version is consistent in all child regions
+      Map<String, ControllerClient> controllerClientMap = getVeniceHelixAdmin().getControllerClientMap(clusterName);
+      List<Callable<Integer>> tasks = new ArrayList<>();
+      controllerClientMap.forEach((region, cc) -> tasks.add(() -> {
+        StoreResponse storeResponse = cc.getStore(storeName, waitingTimeForConsumptionMs);
+        if (storeResponse.isError()) {
+          throw new VeniceException(storeResponse.getError() + " in region " + region);
+        }
+        StoreInfo store = storeResponse.getStore();
+        if (!store.isEnableStoreWrites()) {
+          throw new VeniceException("Unable to rollback since store does not enable write in region " + region);
+        }
+        int futureVersion = getVeniceHelixAdmin().getFutureVersion(clusterName, storeName);
```

Comment:
Is it the right understanding that this line is the only one that's different between this new function and the older `rollbackToBackupVersion` function? Can we extract all of this into a common function, perhaps one which takes a `Function<Admin, Integer>` parameter?

We should update the various comments which talk about backups since this code would be used for both rolling forward and back.

And one more thing... can you double-check that the implementation of `Admin::getFutureVersion` is correct? In the parent, it returns `Store.NON_EXISTING_VERSION`, which does not seem right in the code path here... whereas in the child, it returns:

```
monitor.getTopicsOfOngoingOfflinePushes()
        .stream()
        .filter(topic -> Version.parseStoreFromKafkaTopicName(topic).equals(storeName))
        .findFirst();
```

Which I'm not sure is right either? Does this rely on there being only one future push monitored? Is that guaranteed? In normal operations that should be the case, certainly... but is it not possible that there could be some lingering push that failed or is due to fail but hasn't been cleaned up yet?

Currently, the `Admin::getFutureVersion` API seems to be called only from `StoresRoute::getFutureVersion`, which AFAICT is a read-only API, and so we may or may not have noticed whether it was wonky or not... In any case, we should double-check, since with this PR, it will now be used in a more significant manner as it could mutate the state of the system.

Reply:
Yeah, it is a little wonky.  I have backed out from using it here and now just default to using the most 'future' version in the store list (the biggest numbered one).

Also condensed the code down by using a functional interface so now no more repeated logic.